libs/execution_context/ebpf_core.c

```diff
@@ -423,17 +431,15 @@ _ebpf_core_protocol_resolve_helper(
         goto Done;
     }
 
-    if (count_of_helpers == 0) {
-        goto Done;
-    }
-
-    request_helper_ids = (uint32_t*)ebpf_allocate_with_tag(count_of_helpers * sizeof(uint32_t), EBPF_POOL_TAG_CORE);
-    if (request_helper_ids == NULL) {
-        return_value = EBPF_NO_MEMORY;
-        goto Done;
-    }
-    for (helper_index = 0; helper_index < count_of_helpers; helper_index++) {
-        request_helper_ids[helper_index] = request->helper_id[helper_index];
+    if (count_of_helpers != 0) {
```

Comment:
To check my understanding - in your change, when count_of_helpers is 0 we now need to invoke ebpf_core_resolve_helper() (when we previously didn't) because we need to ensure that ebpf_program_set_program_info_hash() is invoked?